### DISCUSSION TITLE
Show App version when run simulation

### DISCUSSION THREAD

**Discussion by Traiwit** [idaholab/moose#24983] [Q&A General] [answered]
Hi guys,

I have a question regarding showing App version when run simulation.

So when we run moose simulation we will have these lines showing the version of MOOSE, right?

![image](https://github.com/idaholab/moose/assets/79562439/978c9e65-d1b6-4e0f-b441-45c36c076c7a)

BUT I want to show my app version instead of MOOSE version, is this even possible?

![image](https://github.com/idaholab/moose/assets/79562439/1f8c5e51-190b-41c9-b5b2-34260778422c)

![image](https://github.com/idaholab/moose/assets/79562439/ea56ea5f-cc82-4358-8ac2-78a2cc306388)

as you can see they are different

Please let me know if this is possible, if so how to do it? thank you!

Kind regards,
Traiwit

**Comment by GiudGiud** [comment]
Hello

In your header() routine (if it exists yet) in your MooseBehApp.C file you can add something like:
```
  ss << std::setw(25) << "MooseBehApp version: " << MOOSEBEH_REVISION << "\n";
```
this variable is defined in the ./include/base/MooseBehRevision.h as you saw already

If you dont like how it displays I would recommend overriding the MOOSE header (if that s possible, not sure we allow it)

Guillaume

**Reply by Traiwit** [reply]
Hi @GiudGiud,

I thought this would be an easy fix, but apparently it's not.

First of all I didnt have a header() routine yet, so here is what I did:

here is my `moose_behApp.h`

#pragma once

```
#pragma once

#include "MooseApp.h"
#include "MooseBehRevision.h"

class moose_behApp : public MooseApp
{
public:
  static InputParameters validParams();

  moose_behApp(InputParameters parameters);
  virtual ~moose_behApp();



  virtual std::string header() const;
  
  static void registerApps();
  static void registerAll(Factory & f, ActionFactory & af, Syntax & s);
  // static void registerAllObjects(Factory & f, ActionFactory & af, Syntax & s);
};
```

Here is my  header() routine in `moose_behApp.C`

```
std::string
moose_behApp::header() const
{
  std::ostringstream ss;
  ss << std::setw(25) << "MooseBehApp version: " << MOOSEBEH_REVISION << "\n";
  return ss.str();
}
```

it still doesn't print anything

did I do something wrong?

Thanks!
Traiwit

**Reply by GiudGiud** [reply]
this should be it though.
If you look in MooseApp.C you can see the call to header() which you should be overriding

can you add `override` to your function declaration in your moose-behapp.h to make sure it s properly overriding

**Reply by GiudGiud** [reply, marked as answer]
`  virtual std::string header() const override;`

**Reply by Traiwit** [reply]
Awesome @GiudGiud, I got it, though it looks a bit weird

![image](https://github.com/idaholab/moose/assets/79562439/0311f6f2-3820-459b-8f38-6dde81399e0e)

Thank you!

**Reply by GiudGiud** [reply]
ah this isnt great.

Maybe just override the MOOSE routine that prints the  whole header (setupOptions) and pick what you want to print
It s not good practice (if it ends up being a big copy-paste from MOOSE) but it should look as you wish
